src/validation.cpp

```diff
@@ -3974,23 +3974,6 @@ bool ChainstateManager::AcceptBlockHeader(const CBlockHeader& block, BlockValida
     if (ppindex)
         *ppindex = pindex;
 
-    // Since this is the earliest point at which we have determined that a
-    // header is both new and valid, log here.
-    //
-    // These messages are valuable for detecting potential selfish mining behavior;
-    // if multiple displacing headers are seen near simultaneously across many
-    // nodes in the network, this might be an indication of selfish mining. Having
-    // this log by default when not in IBD ensures broad availability of this data
-    // in case investigation is merited.
-    const auto msg = strprintf(
-        "Saw new header hash=%s height=%d", hash.ToString(), pindex->nHeight);
-
-    if (IsInitialBlockDownload()) {
-        LogPrintLevel(BCLog::VALIDATION, BCLog::Level::Debug, "%s\n", msg);
-    } else {
-        LogPrintf("%s\n", msg);
-    }
-
     return true;
```

Comment:
In `master` this logs if `AcceptBlockHeader()` returns `true`. This PR removes the log from here and moves it to the callers.

The call chains are like this:
`ProcessHeadersMessage()` -> `ProcessNewBlockHeaders()` -> `AcceptBlockHeader()` (has the log)
`ProcessMessage()` -> `ProcessNewBlockHeaders()` -> `AcceptBlockHeader()` (has the log)
`submitheader()` -> `ProcessNewBlockHeaders()` -> `AcceptBlockHeader()` (omits the log)
`AcceptBlock()` -> `AcceptBlockHeader()` (omits the log)

are the two log omissions intentional?

Reply:
They're not intentional.

* `submitheader()` is only used for mining, for which this log isn't relevant, so it's fine to omit.
* `AcceptBlock() -> AcceptBlockHeader()`: this is called in the following ways:
  * in `LoadExternalBlockFile()`: not relevant for this log
  * from `ProcessNewBlock()` called from:
    * the kernel demo utility `bitcoin-chainstate.cpp`: not relevant
    * `GenerateBlock()`: (mining / tests)
    * `ProcessBlock()` in net_processing.cpp: 
      * when receiving `CMPCTBLOCK` and in `ProcessCompactBlockTxns()`: part of compact block processing, we've already logged the new header for that earlier
* when receiving a `BLOCK` directly: handled in the third commit

Reply:
I looked at this independently (tracing the call chains before reading this thread), and I came to the same conclusion as Sjors, except for the case of a direct BLOCK, which has an issue that I mentioned below.